test/e2e/tests/request-queuing/ui.spec.js

```diff
@@ -64,6 +64,9 @@ async function openDappAndSwitchChain(driver, dappUrl, chainId) {
 
 async function selectDappClickSendGetNetwork(driver, dappUrl) {
   await driver.switchToWindowWithUrl(dappUrl);
+  // Windows: MetaMask, TestDapp1, TestDapp2
+  const expectedWindowHandles = 3;
+  await driver.waitUntilXWindowHandles(expectedWindowHandles);
   const currentWindowHandles = await driver.getAllWindowHandles();
```

Comment:
the flakiness came from this line: we were calling getAllWindowHandles, expecting to get 3 windows, however sometimes the popup from the previous step was not closed, meaning we could get 4 instead of 3. This made the test fail in the subsequent step where we wait for newWindowHandles which should be 3+1 (or in the case of flakyness 4+1 where this condition is never met)

> const newWindowHandles = await driver.waitUntilXWindowHandles(
    process.env.ENABLE_MV3
      ? currentWindowHandles.length
      : currentWindowHandles.length + 1,
  );